File: main/src/main/java/lavaplayer/source/youtube/YoutubeTrackDetails.java
package lavaplayer.source.youtube;

import lavaplayer.tools.io.HttpInterface;
import lavaplayer.track.AudioTrackInfo;

import java.util.List;

public interface YoutubeTrackDetails {
    AudioTrackInfo getTrackInfo();

    List<YoutubeTrackFormat> getFormats(HttpInterface httpInterface, YoutubeSignatureResolver signatureResolver);

    String getPlayerScript();
}

File: main/src/main/java/lavaplayer/source/youtube/YoutubeTrackDetails.kt
package lavaplayer.source.youtube

import lavaplayer.tools.io.HttpInterface
import lavaplayer.track.AudioTrackInfo

interface YoutubeTrackDetails {
    val trackInfo: AudioTrackInfo
    val playerScript: String?

    fun getFormats(httpInterface: HttpInterface, signatureResolver: YoutubeSignatureResolver): List<YoutubeTrackFormat>
}
